tools/unitctl/unitctl/src/cmd/save.rs

```diff
@@ -0,0 +1,52 @@
+use crate::unitctl::UnitCtl;
+use crate::wait;
+use crate::UnitctlError;
+use crate::requests::send_empty_body_deserialize_response;
+use unit_client_rs::unit_client::UnitClient;
+use tar::{Builder, Header};
+use std::fs::File;
+use std::io::stdout;
+
+
+pub async fn cmd(
+    cli: &UnitCtl,
+    filename: &String
+) -> Result<(), UnitctlError> {
+    if !filename.ends_with(".tar") {
+        eprintln!("Warning: writing tarball to {}", filename);
```

Comment:
We already ask the user to type the name of the tar archive so why is it important to echo it back only in some circumstances?

Reply:
In fact, if you look at the conditional you can see that there is one circumstance where I echo the filename back.
```rust
    if !filename.ends_with(".tar") {
        eprintln!("Warning: writing tarball to {}", filename);
    }
```
The conditional here, `!filename.ends_with(".tar")`, evaluates to true if and only if the filename supplied does not end in tar. The user may have entered a directory name or may have entered something like `.tar.bz` or `.tgz` expecting us to handle compression for them. 

You are asking why this is important? I have yet to work on a software project where the documentation was not overlooked and where the tooling wasn't misused by some users. Our goal here is not to prevent this behavior, but to offer the user all the tools they need to know what our command line tool is going to do with their input.

Reply:
Feels like the error message could be better, but I can't think of an obvious Pareto improvement to it, so I'm inclined to regard it as _good enough._

Where my head bounces around is:

1. _Should_ we automagically handle common compressions?
2. Should the message be more explicit that it's writing an _uncompressed_ tarball?
3. Is it even possible to concisely explain _why_ we're warning the user without making everything more confusing?

The best I can come up with is: `"Warning: uncompressed tarball written to file without .tar extension: {}"`

Or maybe: `"Warning: tarball written to file with unexpected or missing extension: {}"`

Actually getting this right is _hard,_ so leaving it as it stands is probably fine for now. :)

Reply:
> Should we automagically handle common compressions?

I dont think so. Especially if the user needs to insert data into the tarball for certs.

> Should the message be more explicit that it's writing an uncompressed tarball?

Sounds good to me!

> Is it even possible to concisely explain why we're warning the user without making everything more confusing?

I dont know if that is necessary? I think leaving it as is notifies the user of the behavior of the procedure. That was my only goal with this eprint

Reply:
Yeah, here's where I went in my head:

```console
$ unitctl export -f foo
Warning: writing tarball to foo

$ # Bad. What are you *warning* me about?
```

```console
$ unitctl export -f foo.tar.gz
Warning: writing tarball to foo.tar.gz

$ # Bad. Looks like you're warning me that you did what I wanted.
```

```console
$ unitctl export -f foo.tar.gz
Warning: writing uncompressed tarball to foo.tar.gz

$ # Good. Clear and precise, but more coincidental than smart.
```

```console
$ unitctl export -f foo
Warning: writing uncompressed tarball to foo

$ # Okay? Kind of clear, but feels like it could be better. Somehow.
```